.gitlab-ci.yml

```diff
@@ -0,0 +1,76 @@
+# Jobs defined here use the idp/ci docker image from ECR by default. 
+# Images are built via the identity-devops GitLab pipeline.
+
+variables:
+  BUNDLER_VERSION: "2.3.13"
+  ECR_REGISTRY: '${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_REGION}.amazonaws.com'
+
+
+default:
+  image: '${ECR_REGISTRY}/dashboard/ci:latest'
+
+before_script:
+  - gem install bundler --version $BUNDLER_VERSION
+
+.bundle_install: &bundle_install
+  - bundle check || bundle install --retry=3  --path vendor/bundle
+
+ 
+.build_cache:
+  - &ruby_cache
+    key:
+      files:
+        - Gemfile.lock
+    paths:
+      - vendor/bundle
+    policy: pull
+ 
+
+stages:
+  - .pre
+  - test
+  - deploy
+
+
+install_dependencies:
+  stage: .pre
+  variables:
+  cache:
+    - <<: *ruby_cache
+      policy: pull-push
+
+  script:
+    - *bundle_install
+ #   - bundle exec rake login:deploy_json
+
+test_release:
+  stage: test
+  needs:
+    - job: install_dependencies
+  cache:
+    - <<: *ruby_cache
+  script:
+    - *bundle_install
+    - make test
+
+
+deploy_to_cloudgov:
+  only:
+  - main
+  stage: deploy
+  image: '${ECR_REGISTRY}/dashboard/ci:latest'
+  script:
+    - curl  -v -L -o cf8-cli_linux_x86-64.tgz "https://packages.cloudfoundry.org/stable?release=linux64-binary&version=v8&source=github" 
+    - tar -xvzf cf8-cli_linux_x86-64.tgz
+    - mv cf8 /usr/local/bin
+    - cf8 --version
+    - curl -o /usr/share/bash-completion/completions/cf8 https://raw.githubusercontent.com/cloudfoundry/cli-ci/master/ci/installers/completion/cf8
```

Comment:
Following from above, I suspect we'd want to build an image with these rather than install it each time

Reply:
We use this across multiple projects. totaly agree. will create a new issue on the board and work on it